Split UserSendFollowUpNotification into two options

moritz opened the pull request:
At my empolyer, folks aren't happy with the granularity of the follow-up notification settings. Thus this pull request, which separates follow-up notifications for locked, owned ticket from notifications for unlocked tickets in queues/services.

moritz commented:
Should I add the SQL statements to `scripts/DBUpdate-to-5.*.sql`? Or are those generated files?

moritz commented:
Any news or feedback on this front?

NeverMin commented:
i think scripts/DBUpdate-to-4.1.sql

2015-03-13 17:14 GMT+08:00 Moritz Lenz <notifications@github.com>:

> Any news or feedback on this front?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OTRS/otrs/pull/335#issuecomment-78880892>.
>

moritz commented:
I've rebased the branch on current master, and added the SQL to scripts/DBUpdate-to-5.mysql.sql and scripts/DBUpdate-to-5.postgresql.sql. I've left out mssql and oracle, because I have no test databases available.

mgruner commented:
Hi @moritz!
Well, the SQL scripts are handled a little differently: they are automatically generated for all platforms from `scripts/database/upgrade/otrs-upgrate-to-5.xml` via `bin/otrs.xml2sql.pl`. This also means that only simple SQL (see developer manual) is possible that you can specify in the XML. Complex queries will not work and would have to be done in the `DBUpdate-to-5.pl` file, perhaps by using SELECT/INSERT/DELETE in some perl loops.
However, for OTRS 5 we are going to refactor the whole notification handling system. Therefore I'd suggest to close this PR at this point and wait until the new system is in place. Please let me know if this is ok for you.
Regards, mg

mgruner commented:
I'm closing this PR for now. Let's discuss this again later!
